django_mongodb/query.py

```diff
@@ -105,8 +106,28 @@ def join(self, compiler, connection):
         # In the lookup stage, the reference to this column doesn't include
         # the collection name.
         rhs_fields.append(rhs.as_mql(compiler, connection).replace(f"{self.table_name}.", "", 1))
+    extra = self.join_field.get_extra_restriction(self.table_alias, self.parent_alias)
+    if extra:
+        columns = []
+        for expr in extra.leaves():
+            if hasattr(expr, "lhs") and isinstance(expr.lhs, Col):
+                columns.append((expr.lhs, len(lhs_fields)))
+                lhs_fields.append(expr.lhs.as_mql(compiler, connection))
+            if hasattr(expr, "rhs") and isinstance(expr.rhs, Col):
+                columns.append((expr.rhs, None))
+        replacements = {}
+        for col, parent_pos in columns:
+            column_target = col.copy()
+            if column_target.alias == self.table_name:
+                column_target.alias = compiler.collection_name
+            else:
+                column_target.target.db_column = f"{parent_template}{parent_pos}"
+                column_target.target.set_attributes_from_name(f"{parent_template}{len(lhs_fields)}")
```

Comment:
why do we need to use len(lhs_fields) instead of `parent_pos`

Reply:
Fixed